### PR TITLE
Clean up Code Climate rubocop style issues

### PR DESCRIPTION
**Pull request by ABaldwinHunter**
@codeclimate/review

### CODE REVIEW

File: lib/code_climate/test_reporter/git.rb [hunk 1]
```diff
@@ -17,16 +16,16 @@ def branch_from_git_or_ci
...
+          clean = ci_branch.strip.sub(%r{^origin\/}, "")
 
-          clean.size > 0 ? clean : nil
+          !clean.empty? ? clean : nil
         end
 
         def clean_git_branch
           git_branch = String(branch_from_git)
-          clean = git_branch.sub(/^origin\//, "") unless git_branch.start_with?("(")
+          clean = git_branch.sub(%r{^origin\/}, "") unless git_branch.start_with?("(")
```
Comment (maxjacobson) [inline lib/code_climate/test_reporter/git.rb]: I believe the idea with this cop is that using `%r{}` means that `/` no longer delimits the beginning and end of the regular expression, so now you're free to no longer escape `/`'s which appear in the middle of the pattern

File: lib/code_climate/test_reporter/exception_message.rb [hunk 2]
```diff
@@ -18,21 +18,20 @@ def library_name
...
 
       def instructions
-      <<-STR
+        <<-STR
   VCR.configure do |config|
     # your existing configuration
     config.ignore_hosts 'codeclimate.com'
   end
-STR
+        STR
```
Comment (GordonDiggs) [inline lib/code_climate/test_reporter/exception_message.rb]: Do these changes affect the whitespace within the strings here?

File: lib/code_climate/test_reporter/git.rb (hunk 1 above)
Comment (maxjacobson) [inline lib/code_climate/test_reporter/git.rb]: (This, apparently, is called ["leaning toothpick syndrome"](https://en.wikipedia.org/wiki/Leaning_toothpick_syndrome))

Comment (ABaldwinHunter) [inline lib/code_climate/test_reporter/git.rb]: Oo cool. Thanks! Updating

File: lib/code_climate/test_reporter/formatter.rb [hunk 3]
```diff
@@ -39,11 +39,11 @@ def format(result)
       # actually private ...
       def short_filename(filename)
         return filename unless ::SimpleCov.root
-        filename = filename.gsub(/^#{::SimpleCov.root}/, ".").gsub(/^\.\//, "")
+        filename = filename.gsub(/^#{::SimpleCov.root}/, ".").gsub(%r{^./}, "")
```
Comment (maxjacobson) [inline lib/code_climate/test_reporter/formatter.rb]: The `.` probably still needs to be escaped, if it's meant to be a literal `.` and not the regex wildcard

File: lib/code_climate/test_reporter/exception_message.rb (hunk 2 above)
Comment (ABaldwinHunter) [inline lib/code_climate/test_reporter/exception_message.rb]: Nope, I don't believe so. Tested out in IRB:

```console
$ def instructions
     <<-STR
VCR.configure do |config|
  # your existing configuration
     config.ignore_hosts 'codeclimate.com'
 end
     STR
end
=> nil

irb: test = instructions
=> "  VCR.configure do |config|\n    # your existing configuration\n    config.ignore_hosts 'codeclimate.com'\n  end\n"

$ def inst
     <<-STR
VCR.configure do |config|
  # your existing configuration
     config.ignore_hosts 'codeclimate.com'
   end
STR
    end
=> nil
irb: test_two = inst
=> "  VCR.configure do |config|\n    # your existing configuration\n    config.ignore_hosts 'codeclimate.com'\n  end\n"
irb(main):028:0> test == test_two
=> true
```

File: lib/code_climate/test_reporter/formatter.rb (hunk 3 above)
Comment (ABaldwinHunter) [inline lib/code_climate/test_reporter/formatter.rb]: Thanks :) good catch

**Comment by maxjacobson** [comment]
LGTM

File: lib/code_climate/test_reporter/calculate_blob.rb [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
   module TestReporter
     class CalculateBlob
-
       def initialize(file_path)
         @file_path = file_path
       end
 
       def blob_id
-        calculate_with_file or calculate_with_git
+        calculate_with_file || calculate_with_git
```
Comment (maxjacobson) [inline lib/code_climate/test_reporter/calculate_blob.rb]: ah, one last note: we recently updated our styleguide to allow this kind of usage of `or` (https://github.com/codeclimate/styleguide/commit/3aa6d71925acad7b877f7afd5a6aee971f9df332) so might be worth considering leaving this one, but up to you

Comment (ABaldwinHunter) [inline lib/code_climate/test_reporter/calculate_blob.rb]: Oo I like that. Updated. Thanks!

**Pull request merged**